deploy-lvmo/deploy.sh

```diff
@@ -0,0 +1,110 @@
+    else
+        envsubst <${SOURCE_FILE} >${DESTINATION_FILE}
+    fi
+}
+
+function extract_vars() {
+    # Extract variables from config file
+    DISKS_PATH=${1}
+    raw_disks=$(yq eval "${DISKS_PATH}" "${EDGECLUSTERS_FILE}" | sed s/null//)
+    disks=$(echo ${raw_disks} | tr -d '\ ' | sed 's/-/,/g' | sed 's/,*//' | sed 's/,*//')
+    disks_count=$(echo ${disks} | sed 's/,/\n/g' | wc -l)
+
+    for node in $(oc --kubeconfig=${EDGE_KUBECONFIG} get nodes -o name | cut -f2 -d/); do
+        nodes+="${node},"
+    done
+
+    nodes=$(echo ${nodes%*,})
+
+    # Final Variables
+    export CHANGEME_NODES="[${nodes}]"
+    export CHANGEME_DEVICES="[${disks}]"
+    export CHANGEME_STORAGE_DEVICE_SET_COUNT="${disks_count}"
+}
+
+function extract_kubeconfig() {
+    ## Extract the Edge-cluster kubeconfig and put it on the shared folder
+    export EDGE_KUBECONFIG=${OUTPUTDIR}/kubeconfig-${1}
+    oc --kubeconfig=${KUBECONFIG_HUB} extract -n $edgecluster secret/$edgecluster-admin-kubeconfig --to - >${EDGE_KUBECONFIG}
+}
+
+# Load common vars
+source ${WORKDIR}/shared-utils/common.sh
+if ! ./verify.sh; then
+    echo ">>>> Modify files to replace with pipeline info gathered"
+    echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+    sed -i "s/CHANGEME/$OC_ODF_VERSION/g" manifests/03-LVMO-Subscription.yaml
+
+    if [[ -z ${ALLEDGECLUSTERS} ]]; then
+        ALLEDGECLUSTERS=$(yq e '(.edgeclusters[] | keys)[]' ${EDGECLUSTERS_FILE})
+    fi
+
+    index=0
+    for edgecluster in ${ALLEDGECLUSTERS}; do
+	NUM_M=$(yq e ".edgeclusters[${index}].[]|keys" ${EDGECLUSTERS_FILE} | grep master | wc -l | xargs)
+
+        echo ">>>> Deploy manifests to install LSO and LocalVolume: ${edgecluster}"
+        echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+        echo "Extract Kubeconfig for ${edgecluster}"
+        extract_kubeconfig ${edgecluster}
+        echo "Filling vars for ${edgecluster}"
+        extract_vars ".edgeclusters[].${edgecluster}.master0.storage_disk"
+        oc --kubeconfig=${EDGE_KUBECONFIG} apply -f manifests/01-Namespace.yaml
+        sleep 2
+        oc --kubeconfig=${EDGE_KUBECONFIG} apply -f manifests/02-LVMO-OperatorGroup.yaml
+        sleep 2
+        oc --kubeconfig=${EDGE_KUBECONFIG} apply -f manifests/03-LVMO-Subscription.yaml
+        sleep 5
+
+        echo ">>>> Waiting for subscription and crd on: ${edgecluster}"
+        echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>"
+        declare -a LVMOCRDS=("lvmclusters.lvm.topolvm.io" "lvmvolumegroupnodestatuses.lvm.topolvm.io" "lvmvolumegroups.lvm.topolvm.io")
+        for crd in ${LVMOCRDS[@]}; do
+            check_resource "crd" "${crd}" "Established" "openshift-local-storage" "${EDGE_KUBECONFIG}"
+        done
+
+        echo ">>>> Render and apply manifest to deploy LVMCluster"
+        echo ">>>>>>>>>>>>>>>>>>>>>>>>>>>"
+        render_file manifests/04-LVMO-LVMOCluster.yaml
```

Comment:
seems that apply the manifest 04-LVMO-LVMOCluster.yaml is missing in here?

Reply:
I think the apply is on line 18 (envsubst <${SOURCE_FILE} | oc --kubeconfig=${EDGE_KUBECONFIG} apply -f -) 
So this is OK